main_test.go

```diff
@@ -0,0 +1,291 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	testConsent "identity_platform_login_ui/ory_mocking/Consent"
+	testErrors "identity_platform_login_ui/ory_mocking/Errors"
+	testLoginUpdate "identity_platform_login_ui/ory_mocking/Login"
+	testLoginBrowser "identity_platform_login_ui/ory_mocking/LoginBrowser"
+	testServers "identity_platform_login_ui/ory_mocking/Testservers"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// --------------------------------------------
+// TESTING WITH CORRECT SERVERS
+// --------------------------------------------
+func TestHandleCreateFlowWithoutCookie(t *testing.T) {
```

Comment:
`testing.T` provides `Cleanup` function (https://pkg.go.dev/testing#T.Cleanup). You could pass it to `testServers.CreateTestServers` and the servers can then register their own clean up methods so that you don't have to do it in the test.

Reply:
I did it for most tests. The error tests are "generated" with a function, which I would rather not pass the t test object to for now.

Reply:
You should pass `t` as a parameter to the test server initialization function, this will allow the server to take care of its own cleanup:
```go 
...
func TestHandleCreateFlowWithoutCookie(t *testing.T) {
        testServers.CreateTestServers(t)
        ...
}
...

func CreateTestServers() func(t *testing.T) {
	kratos = createKratosMockServer(t)
	hydra = createHydraMockServer(t)
}
...

func createKratosMockServer(t *testing.T) *httptest.Server {
       ...
       t.Cleanup(s.Close())
       return s
}
```

Reply:
done. at first I was more in favour of restricting the testing objects to the test_main module, but your solution is valid too. I changed my fork to conform to this.